qualtran/bloqs/gf_arithmetic/gf2_inverse.py

```diff
@@ -96,60 +105,111 @@ def signature(self) -> 'Signature':
     def qgf(self) -> QGF:
         return QGF(characteristic=2, degree=self.bitsize)
 
+    @cached_property
+    def n_junk_regs(self) -> SymbolicInt:
+        return 2 * bit_length(self.bitsize - 1) + self.bitsize_hamming_weight
+
+    @cached_property
+    def bitsize_hamming_weight(self) -> SymbolicInt:
+        """Hamming weight of self.bitsize - 1"""
+        return (
+            bit_length(self.bitsize - 1)
+            if is_symbolic(self.bitsize)
+            else int(self.bitsize - 1).bit_count()
+        )
+
     def my_static_costs(self, cost_key: 'CostKey'):
+        from qualtran._infra.gate_with_registers import total_bits
         from qualtran.resource_counting import QubitCount
 
         if isinstance(cost_key, QubitCount):
-            return self.signature.n_qubits()
+            return total_bits(self.signature.rights())
 
         return NotImplemented
 
-    def build_composite_bloq(self, bb: 'BloqBuilder', *, x: 'Soquet') -> Dict[str, 'Soquet']:
+    def build_composite_bloq(self, bb: 'BloqBuilder', *, x: 'Soquet') -> Dict[str, 'SoquetT']:
         if is_symbolic(self.bitsize):
             raise DecomposeTypeError(f"Cannot decompose symbolic {self}")
+
         result = bb.allocate(dtype=self.qgf)
         if self.bitsize == 1:
             x, result = bb.add(GF2Addition(self.bitsize), x=x, y=result)
             return {'x': x, 'result': result}
 
-        x = bb.add(GF2Square(self.bitsize), x=x)
-        x, result = bb.add(GF2Addition(self.bitsize), x=x, y=result)
-
         junk = []
-        for i in range(2, self.bitsize):
-            x = bb.add(GF2Square(self.bitsize), x=x)
-            x, result, new_result = bb.add(GF2Multiplication(self.bitsize), x=x, y=result)
-            junk.append(result)
-            result = new_result
-        x = bb.add(GF2Square(self.bitsize), x=x)
-        return {'x': x, 'result': result} | ({'junk': np.array(junk)} if junk else {})
+        beta = bb.allocate(dtype=self.qgf)
```

Comment:
```suggestion
        # The algorithm is descriped on page 4 and 5 of https://arxiv.org/abs/1209.6348 and resembles binary exponentiation.
        # The inverse is computed as (B_{n-1})^2. Where B_1 = x and B_{i+j} = B_i B_j^{2^i}.
        beta = bb.allocate(dtype=self.qgf)
```

Reply:
cool, yeah. I'd see if this could make its way to the class docstring so it shows up in the docs

Reply:
Added to the class docstring and regenerated the docs